lib/visual_metrics.py

```diff
@@ -285,35 +288,51 @@ def ListOneTarget(codecs, rate, videofile, do_score, datatable,
     # Ignore results that score less than zero.
     if my_optimizer.Score(bestsofar) < 0.0:
       return
-    (datatable.setdefault(codec_name, {})
-             .setdefault(videofile.basename, [])
-             .append((bestsofar.result['bitrate'], bestsofar.result['psnr'])))
```

Comment:
Please add a short comment here that describes datatable; it's hard to understand these setdefault and append chains.